Lombiq.Tests.UI/Services/UITestExecutor.cs

```diff
@@ -51,15 +51,16 @@ public static class UITestExecutor
 
         configuration.TestOutputHelper.WriteLineTimestampedAndDebug("Finished preparation for {0}.", testManifest.Name);
 
+        await _numberOfTestsLimitLock.WaitAsync();
+
         if (_numberOfTestsLimit == null && configuration.MaxParallelTests > 0)
         {
-            lock (_numberOfTestsLimitLock)
-            {
-                _numberOfTestsLimit ??= new SemaphoreSlim(configuration.MaxParallelTests);
-            }
+            _numberOfTestsLimit = new SemaphoreSlim(configuration.MaxParallelTests);
         }
```

Comment:
[InferSharp ](https://github.com/microsoft/infersharp) reported a thread safety violation here, I can agree with it.
![image](https://user-images.githubusercontent.com/15349881/227495008-2df753c0-f189-4e9d-b8e3-eb5ae261e9f8.png)
Description: Read/Write race. Non-private method `UITestExecutor.ExecuteOrchardCoreTestAsync(...)` reads without synchronization from `Services.UITestExecutor.Lombiq.Tests.UI.Services.UITestExecutor._numberOfTestsLimit`. Potentially races with write in method `UITestExecutor.ExecuteOrchardCoreTestAsync(...)`. Reporting because this access may occur on a background thread.

Reply:
Could you please fix this?

Reply:
I fixed it already, just noted why I changed something out of scope.

Reply:
Ah, great! BTW: https://github.com/Lombiq/GitHub-Actions/issues/219

Reply:
Agree